File: C++11/151.cpp
#include <iostream>

using namespace std;

int main() {
    char m1, m2, l1, l2;
    cin >> m1 >> m2;
    cin >> l1 >> l2;

    bool flag;
    if (m1 - l1 == (char)-10 || m1 - l1 == (char)-7 || m1 - l1 == (char)17) {
        if (m1 - l2 == (char)-10 || m1 - l2 == (char)-7 || m1 - l2 == (char)17 || m1 == l2) cout << "MING" << endl;
        else {
            if (m2 - l2 == (char)-10 || m2 - l2 == (char)-7 || m2 - l2 == (char)17) cout << "MING" << endl;
            else cout << "LIHUA" << endl;
        }
    } else if (m1 == l1) {
        if (m2 - l2 == (char)-10 || m2 - l2 == (char)-7 || m2 - l2 == (char)17) cout << "MING" << endl;
        else if (m2 == l2) cout << "TIE" << endl;
        else cout << "LIHUA" << endl;
    } else {
        if (m2 - l1 == (char)-10 || m2 - l1 == (char)-7 || m2 - l1 == (char)17) {
            if (m2 - l2 == (char)-10 || m2 - l2 == (char)-7 || m2 - l2 == (char)17 || m2 == l2) cout << "MING" << endl;
            else cout << "LIHUA" << endl;
        } else {
            cout << "LIHUA" << endl;
        }
    }

    return 0;   
}

File: Python3/151.py
m1, m2 = input().split()
l1, l2 = input().split()
m1 = ord(m1)
m2 = ord(m2)
l1 = ord(l1)
l2 = ord(l2)

if m1 - l1 == -10 or m1 - l1 == -7 or m1 - l1 == 17:
    if m1 - l2 == -10 or m1 - l2 == -7 or m1 - l2 == 17 or m1 == l2:
        print("MING")
    else :
        if m2 - l2 == -10 or m2 - l2 == -7 or m2 - l2 == 17:
            print("MING")
        else :
            print("LIHUA")
elif m1 == l1:
    if m2 - l2 == -10 or m2 - l2 == -7 or m2 - l2 == 17:
        print("MING")
    elif m2 == l2 :
        print("TIE")
    else :
        print("LIHUA")
else :
    if m2 - l1 == -10 or m2 - l1 == -7 or m2 - l1 == 17:
        if m2 - l2 == -10 or m2 - l2 == -7 or m2 - l2 == 17 or m2 == l2:
            print("MING")
        else :
            print("LIHUA")
    else :
        print("LIHUA")